src/qibo/models/circuit.py

```diff
@@ -88,24 +85,30 @@ def from_fused(self):
                 queue.append(gate.gates[0])
         return queue
 
-    def append(self, gate: gates.Gate):
-        super().append(gate)
-        if gate.qubits:
-            qubits = gate.qubits
-        else:  # special gate acting on all qubits
-            qubits = tuple(range(self.nqubits))
-
-        if isinstance(gate, gates.M):
-            self.nmeasurements += 1
-
-        # calculate moment index for this gate
-        idx = max(self.moment_index[q] for q in qubits)
-        for q in qubits:
-            if idx >= len(self.moments):
-                # Add a moment
-                self.moments.append(len(self.moments[-1]) * [None])
-            self.moments[idx][q] = gate
-            self.moment_index[q] = idx + 1
+    @property
+    def nmeasurements(self):
+        return len([1 for gate in self if isinstance(gate, gates.M)])
```

Comment:
```suggestion
        return len(self.gates_of_type(gates.M))
```

Reply:
Is this possible? I think `_Queue` doesn't have a `gates_of_type` method. It may still be possible to avoid the `for` loop using [`filter`](https://docs.python.org/3/library/functions.html#filter).